For musl libc return the file position using
ftell(3) because fpos_t is opaque.

--- 3rdparty/hts_engine_API/lib/HTS_misc.c	2015-02-18 21:10:05.000000000 +0100
+++ 3rdparty/hts_engine_API/lib/HTS_misc.c	2015-10-31 17:33:53.946349145 +0100
@@ -247,8 +247,10 @@
       fgetpos((FILE *) fp->pointer, &pos);
 #if defined(_WIN32) || defined(__CYGWIN__) || defined(__APPLE__) || defined(__ANDROID__) || defined(__FreeBSD__)
       return (size_t) pos;
-#else
+#elif defined(__GLIBC__)
       return (size_t) pos.__pos;
+#else
+      return (size_t) ftell(fp->pointer);
 #endif                          /* _WIN32 || __CYGWIN__ || __APPLE__ || __ANDROID__ */
    }
    HTS_error(0, "HTS_ftell: Unknown file type.\n");
